cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -240,25 +237,21 @@ public String commitAndPush(final CommitMeta commitMeta,
           "No changes have been made for the applications to push.");
     }
 
-    git.add().addFilepattern(fileChanged.toString()).call();
-
-    RevCommit commit = getCommitCommand(commitMeta).call();
-
-    String fileHash;
-    try {
-      fileHash = getFileHash(fileChanged, commit);
-    } catch (IOException e) {
-      throw new GitOperationException(
-          String.format("Failed to get fileHash for %s", fileChanged),
-          e);
+    List<T> output = new LinkedList<>();
```

Comment:
should use an ArrayList. It basically always performs better than a LinkedList, even if it needs to get resized underneath. This is especially true when we know the size beforehand (like we do here) and can pass it in to the constructor.

Reply:
Updated.